src/main/java/com/iota/iri/network/impl/TransactionRequesterWorkerImpl.java

```diff
@@ -1,183 +0,0 @@
-
-        return this;
-    }
-
-    /**
-     * {@inheritDoc}
-     * <p>
-     * To reduce the overhead for the node we only trigger this worker if the request queue gets bigger than the {@link
-     * #REQUESTER_THREAD_ACTIVATION_THRESHOLD}. Otherwise we rely on the processing of the queue due to normal outgoing
-     * traffic like transactions that get relayed by our node.
-     * </p>
-     */
-    @Override
-    public boolean processRequestQueue() {
-        try {
-            if (isActive()) {
-                TransactionViewModel transaction = getTransactionToSendWithRequest();
-                if (isValidTransaction(transaction)) {
-                    sendToNodes(transaction);
-                    return true;
-                }
-            }
-        } catch (Exception e) {
-            log.error("unexpected error while processing the request queue", e);
-        }
-        return false;
-    }
-
-    private void sendToNodes(TransactionViewModel transaction) {
-        for (Neighbor neighbor : neighborRouter.getConnectedNeighbors().values()) {
-            try {
-                // automatically adds the hash of a requested transaction when sending a packet
-                neighborRouter.gossipTransactionTo(neighbor, transaction);
-            } catch (Exception e) {
-                log.error("unexpected error while sending request to neighbour", e);
-            }
-        }
-    }
-
-    @VisibleForTesting
-    boolean isActive() {
-        return transactionRequester.numberOfTransactionsToRequest() >= REQUESTER_THREAD_ACTIVATION_THRESHOLD;
-    }
-
-    @VisibleForTesting
-    boolean isValidTransaction(TransactionViewModel transaction) {
-        return transaction != null && (
-                transaction.getType() != TransactionViewModel.PREFILLED_SLOT
-             || transaction.getHash().equals(Hash.NULL_HASH));
-    }                                     
-
-    @Override
-    public void start() {
-        executorService.silentScheduleWithFixedDelay(this::processRequestQueue, 0, REQUESTER_THREAD_INTERVAL,
-                TimeUnit.MILLISECONDS);
-    }
-
-    @Override
-    public void shutdown() {
-        executorService.shutdownNow();
-    }
-
-    /**
-     * <p>
-     * Retrieves a random solid tip that can be sent together with our request.
-     * </p>
-     * </p>
-     * It retrieves the hash of the tip from the {@link #tipsViewModel} and tries to load it from the
-     * database.
-     * </p>
-     *
-     * @return a random tip
-     * @throws Exception if anything unexpected happens while trying to retrieve the random tip.
-     */
-    @VisibleForTesting
-    TransactionViewModel getTransactionToSendWithRequest() throws Exception {
-        Hash tip = tipsViewModel.getRandomSolidTipHash();
-        if (tip == null) {
-            tip = tipsViewModel.getRandomNonSolidTipHash();
-        }
-
-        return TransactionViewModel.fromHash(tangle, tip == null ? Hash.NULL_HASH : tip);
-    }
-}
```

Comment:
Yay!!
Patchy worker is gone :-)

Reply:
![](https://media.giphy.com/media/5xaOcLwEvFOizxHVyVy/giphy.gif)